libkbfs/favorites.go

```diff
@@ -107,6 +112,87 @@ func NewFavorites(config Config) *Favorites {
 	return newFavoritesWithChan(config, make(chan *favReq, 100))
 }
 
+func (f *Favorites) readCacheFromDisk(ctx context.Context) error {
+	// Read the encrypted cache from disk
+	folder := f.config.StorageRoot()
+	subfolder := "kbfs.favorites.cache"
+	session, err := f.config.KBPKI().GetCurrentSession(ctx)
+	if err != nil {
+		return err
+	}
+	user := string(session.UID)
+	path := filepath.Join(folder, subfolder, user)
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	// Send the data to the service to be decrypted
+	decryptedData, err := f.config.KeybaseService().DecryptFavorites(ctx, data)
+	if err != nil {
+		return err
+	}
+
+	// Decode the data into the a map
+	decoder := gob.NewDecoder(bytes.NewBuffer(decryptedData))
+	var cache map[Favorite]bool
+	err = decoder.Decode(&cache)
+	if err != nil {
+		return err
+	}
+
+	f.cache = cache
+	return nil
+}
+
+func (f *Favorites) writeCacheToDisk(ctx context.Context) error {
+	// Encode the cache map into a byte buffer
+	buf := &bytes.Buffer{}
+	encoder := gob.NewEncoder(buf)
+	err := encoder.Encode(f.cache)
+	if err != nil {
+		return err
+	}
+
+	// Send the byte buffer to the service for encryption
+	data, err := f.config.KeybaseService().EncryptFavorites(ctx, buf.Bytes())
+	if err != nil {
+		return err
+	}
+
+	// Write the encrypted cache to disk
+	folder := f.config.StorageRoot()
+	subfolder := "kbfs.favorites.cache"
+	session, err := f.config.KBPKI().GetCurrentSession(ctx)
+	if err != nil {
+		return err
+	}
+	user := string(session.UID)
+	path := filepath.Join(folder, subfolder, user)
+	// TODO: is this the right permissions
+	return ioutil.WriteFile(path, data, 0600)
+}
+
+// InitForTest starts the Favorites cache's internal processing loop without
+// loading cached favorites from disk.
+func (f *Favorites) InitForTest() {
+	go f.loop()
+}
+
+// Initialize loads the favorites cache from disk and starts listening for
+// requests asynchronously.
+func (f *Favorites) Initialize(ctx context.Context) {
+	// load cache from disk
+	err := f.readCacheFromDisk(ctx)
+	if err != nil {
+		f.config.MakeLogger("favorites").CDebugf(nil,
```

Comment:
Maybe `CWarningf` instead?

Reply:
We do expect this to happen on the first login of each user; does that qualify for a warning?

Reply:
You mean we expect it to happen _before_ the first login of each user?  In that case, debug is fine.